spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -694,6 +726,71 @@ else if (isChannelTransacted()) {
 		}, obtainTargetConnectionFactoryIfNecessary(this.receiveConnectionFactorySelectorExpression, queueName));
 	}
 
+	/**
+	 * Blocking receive with timeout.
+	 * @param queueName the queue to receive from.
+	 * @return The message, or null if the time expires.
+	 * @since 1.5
+	 */
+	protected Message doReceive(final String queueName) {
+		return execute(new ChannelCallback<Message>() {
+
+			@Override
+			public Message doInRabbit(Channel channel) throws Exception {
+				channel.basicQos(1);
+				final CountDownLatch latch = new CountDownLatch(1);
+				QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+					@Override
+					public void handleCancel(String consumerTag) throws IOException {
+						super.handleCancel(consumerTag);
+						latch.countDown();
+					}
+
+					@Override
+					public void handleConsumeOk(String consumerTag) {
+						super.handleConsumeOk(consumerTag);
+						latch.countDown();
+					}
+
+				};
+				channel.basicConsume(queueName, consumer);
+				if (!latch.await(10, TimeUnit.SECONDS)) {
+					if (channel instanceof ChannelProxy) {
+						((ChannelProxy) channel).getTargetChannel().close();
+					}
+					throw new AmqpException("Blocking receive, consumer failed to consume");
+				}
+				Delivery delivery;
+				if (receiveTimeout < 0) {
+					delivery = consumer.nextDelivery();
+				}
+				else {
+					delivery = consumer.nextDelivery(receiveTimeout);
+				}
+				channel.basicCancel(consumer.getConsumerTag());
```

Comment:
`basicCancel` to prevent the consumption for the next message ? 

From other side can't we use shared `QueueingConsumer` instance since this operation is blocked?
Or at least lazy-load (TTL removal) per `queueName`, but blocked anyway...